iChor/iChorCNA_hg19/iChorCNA_hg19.wdl

```diff
@@ -1,327 +1,352 @@
         File cna = ichorCNATask.cna
         File segTxt = ichorCNATask.segTxt
         File seg = ichorCNATask.seg
         File rdata = ichorCNATask.rdata
 
-		Float tumor_fraction = extractIchorParams.tumor_fraction
-		Float ploidy = extractIchorParams.ploidy
-		String subclone_fraction = extractIchorParams.subclone_fraction
-		String fraction_genome_subclonal = extractIchorParams.fraction_genome_subclonal
-		String fraction_cna_subclonal = extractIchorParams.fraction_cna_subclonal
-		String gc_map_correction_mad = extractIchorParams.gc_map_correction_mad
+        Float tumor_fraction = extractIchorParams.tumor_fraction
+        Float ploidy = extractIchorParams.ploidy
+        String subclone_fraction = extractIchorParams.subclone_fraction
+        String fraction_genome_subclonal = extractIchorParams.fraction_genome_subclonal
+        String fraction_cna_subclonal = extractIchorParams.fraction_cna_subclonal
+        String gc_map_correction_mad = extractIchorParams.gc_map_correction_mad
         Int top_solution_log_likelihood = extractIchorParams.top_solution_log_likelihood
 
-		File bias = ichorCNATask.bias
-		File tpdf = ichorCNATask.tpdf
-		File correct = ichorCNATask.correct
-		File params = ichorCNATask.params
+        File bias = ichorCNATask.bias
+        File tpdf = ichorCNATask.tpdf
+        File correct = ichorCNATask.correct
+        File params = ichorCNATask.params
 
-		File optimalSolution = ichorCNATask.optimalSolution
-		File outSolutions = ichorCNATask.outSolutions
-		File perChromosomePlots = bundlePerChromosomePlots.output_plot
+        File optimalSolution = ichorCNATask.optimalSolution
+        File outSolutions = ichorCNATask.outSolutions
+        File perChromosomePlots = bundlePerChromosomePlots.output_plot
     }
 }
 
 task read_counter {
-	input {
-		File bam_file
-		File bam_index
-		String chrs
-		String sample_id
-		Int bin_size
-
-		Int min_qual = 20
-		Int memGB = 4
-		Int diskGB = 50
-		Int preemptible = 2
-
-		String? docker_override
-	}
-
-	command {
-		ln -vs ${bam_index} ${sample_id}.bam.bai
-		ln -vs ${bam_file} ${sample_id}.bam
-
-		/HMMcopy/bin/readCounter ${sample_id}.bam -c ${chrs} -w ${bin_size}  \
-		-q ${min_qual} > ${sample_id}.bin${bin_size}.wig
-	}
-
-	runtime {
-		docker: select_first([docker_override, "us.gcr.io/tag-team-160914/bloodbiopsy-hmmcopy:0.0.1"])
-		disks: "local-disk ${diskGB} HDD"
-		memory: "${memGB}GB"
-		preemptible: "${preemptible}"
-		maxRetries: 1
-	}
-
-	output {
-		File wig_file = "${sample_id}.bin${bin_size}.wig"
-	}
+    input {
+        File bam_file
+        File bam_index
+        String chrs
+        String sample_id
+        Int bin_size
+
+        Int min_qual = 20
+        Int memGB = 4
+        Int diskGB = 50
+        Int preemptible = 2
+
+        String docker_override = "us.gcr.io/tag-team-160914/bloodbiopsy-hmmcopy:0.0.1"
+    }
+    
+    String wig_file_name = "~{sample_id}.bin~{bin_size}.wig"
+
+    command <<<
+        set -euxo pipefail
```

Comment:
These `set` calls I could make an argument are dangerous in that it's the one change (I **think** the one change at least) that could in theory change behavior. OTOH it's obviously an improvement.

As a former member of DSP methods: how likely do you think such a change would trigger notably more intense testing before acceptance?

Reply:
Honestly that kind of thing is all over the map. That said, in this particular workflow I'm hard-pressed to find any task that could have a bash error and be okay to continue. Most are basically simple front-matter with a 1-liner payload. If the R-script / python / gs fails the task should fail.

Reply:
Fair